frida-gum/src/lib.rs

```diff
@@ -121,6 +124,18 @@ impl NativePointer {
     }
 }
 
+impl From<&NativePointer> for *mut c_void {
+    fn from(other: &NativePointer) -> Self {
+        other.0
+    }
+}
+
+impl From<NativePointer> for *mut c_void {
+    fn from(other: NativePointer) -> Self {
+        other.0
+    }
+}
+
```

Comment:
The fact that we marked the `c_void` as `pub` was a mistake in the API in the first place in my opinion, we should have abstracted the `NativePointer` in a way that prevents these conversions. This is fine because I think eventually I'll refactor this anyways and get rid of this strange API.

Reply:
Given the nature of Fria and what it's used for, I think being able to get the pointer as a `usize` ergonomically is important for printing, doing pointer arithmetics, etc. What's your take on that? Do you think we should implement the arithmetic operations on `NativePointer` (e.g. `Add<Rhs=usize>`)?